### DISCUSSION TITLE
Is it possible to include ATLAS and OpenMPI into cmakefiles?

### DISCUSSION THREAD

**Discussion by TangLaoya** [scivision/mumps-superbuild#49] [Ideas]
Dear all,
Since the MUMPS used all open source code to build the project, can we also include ATLAS and OpenMPI into cmake, at least, we can take them as options.

Thanks,
Tang Laoya

**Comment by scivision** [comment]
Are you wanting to build them or? As it is, the Mumps CMake finds OpenMPI and should work with finding ATLAS

**Reply by TangLaoya** [reply]
Thanks. I mean let the cmake use git to download these if users want to do.

**Reply by scivision** [reply]
ATLAS has been several years since last release and it has trouble building on modern systems. It takes numerous patches and I don't have the time to do this: https://sources.debian.org/patches/atlas/3.10.3-13/

 OpenMPI I have in another project: https://github.com/gemini3d/external/blob/main/cmake/openmpi.cmake
 This script could be copied and made into your own project. I have several hundred projects and trying not to make more.
